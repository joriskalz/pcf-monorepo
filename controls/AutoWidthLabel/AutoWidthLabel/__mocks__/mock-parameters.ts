/* istanbul ignore file */

import { IInputs } from "../generated/ManifestTypes";
import {
  MockDecimalNumberProperty,
  MockEnumProperty,
  MockStringProperty,
  MockWholeNumberProperty,
} from "./mock-context";

export function getMockParameters(): IInputs {
  return {
    Text: new MockStringProperty(),
    PaddingLeft: new MockWholeNumberProperty(),
    PaddingRight: new MockWholeNumberProperty(),
    PaddingTop: new MockWholeNumberProperty(),
    PaddingBottom: new MockWholeNumberProperty(),
    FontName: new MockStringProperty(),
    FontSize: new MockDecimalNumberProperty(),
    FontSizeUnits: new MockEnumProperty("1"),
    FontColor: new MockStringProperty(),
    FontWeight: new MockStringProperty(),
    DisabledFontColor: new MockStringProperty(),
    DisabledFontWeight: new MockStringProperty(),
    FocusFontColor: new MockStringProperty(),
    FocusFontWeight: new MockStringProperty(),
    HoverFontColor: new MockStringProperty(),
    HoverFontWeight: new MockStringProperty(),
    FillColor: new MockStringProperty(),
    DisabledFillColor: new MockStringProperty(),
    FocusFillColor: new MockStringProperty(),
    HoverFillColor: new MockStringProperty(),
    //DisabledHoverFillColor: new MockStringProperty(),
    BorderColor: new MockStringProperty(),
    BorderThickness: new MockWholeNumberProperty(),
    BorderRadius: new MockWholeNumberProperty(),
    DisabledBorderColor: new MockStringProperty(),
    FocusBorderColor: new MockStringProperty(),
    FocusBorderThickness: new MockWholeNumberProperty(),
    HoverBorderColor: new MockStringProperty(),
    HoverBorderThickness: new MockWholeNumberProperty(),
    //RenderBorderStyle: new MockEnumProperty<"0" | "1">(),
  };
}
