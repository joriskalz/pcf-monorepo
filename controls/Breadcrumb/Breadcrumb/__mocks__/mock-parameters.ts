/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockStringProperty, MockWholeNumberProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        Theme: new MockStringProperty(),
        items: new MockDataSet([]),
        MaxDisplayedItems: new MockWholeNumberProperty(),
        OverflowIndex: new MockWholeNumberProperty(),
    };
}
