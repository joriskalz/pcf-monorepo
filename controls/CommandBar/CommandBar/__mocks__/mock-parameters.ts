/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockStringProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        Theme: new MockStringProperty(),
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        items: new MockDataSet([]),
    };
}
