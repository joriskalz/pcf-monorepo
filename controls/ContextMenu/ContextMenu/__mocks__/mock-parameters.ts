/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockEnumProperty, MockStringProperty, MockTwoOptionsProperty, MockWholeNumberProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        Theme: new MockStringProperty(),
        TextAlignment: new MockEnumProperty(),
        items: new MockDataSet([]),
        Chevron: new MockTwoOptionsProperty(),
        FillColor: new MockStringProperty(),
        BorderColor: new MockStringProperty(),
        BorderRadius: new MockWholeNumberProperty(),
        FontColor: new MockStringProperty(),
        FontSize: new MockWholeNumberProperty(),
        HoverBorderColor: new MockStringProperty(),
        HoverFillColor: new MockStringProperty(),
        HoverFontColor: new MockStringProperty(),
        HoverIconColor: new MockStringProperty(),
        IconColor: new MockStringProperty(),
        IconSize: new MockWholeNumberProperty(),
    };
}
