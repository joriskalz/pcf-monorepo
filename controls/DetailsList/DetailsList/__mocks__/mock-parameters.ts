/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockEnumProperty, MockStringProperty, MockTwoOptionsProperty, MockWholeNumberProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        Theme: new MockStringProperty(),
        Compact: new MockTwoOptionsProperty(),
        CurrentSortColumn: new MockStringProperty(),
        CurrentSortDirection: new MockEnumProperty('0'),
        PageSize: new MockWholeNumberProperty(),
        RaiseOnRowSelectionChangeEvent: new MockTwoOptionsProperty(),
        SelectRowsOnFocus: new MockTwoOptionsProperty(),
        SelectionType: new MockEnumProperty('0'),
        AlternateRowColor: new MockStringProperty(),
        HeaderVisible: new MockTwoOptionsProperty(false),
        SelectionAlwaysVisible: new MockTwoOptionsProperty(false),
        records: new MockDataSet([]),
        columns: new MockDataSet([]),
        LargeDatasetPaging: new MockTwoOptionsProperty(),
    };
}
