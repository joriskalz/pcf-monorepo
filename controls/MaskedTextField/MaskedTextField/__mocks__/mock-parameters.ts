/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockStringProperty, MockTwoOptionsProperty } from './mock-context';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        Theme: new MockStringProperty(),
        Value: new MockStringProperty(),
        Prefix: new MockStringProperty(),
        Suffix: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        ErrorMessage: new MockStringProperty(),
        Mask: new MockStringProperty(),
        MaskFormat: new MockStringProperty(),
        DelayOutput: new MockTwoOptionsProperty(),
    };
}
