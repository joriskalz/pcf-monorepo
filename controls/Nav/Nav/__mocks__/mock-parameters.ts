/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockStringProperty, MockTwoOptionsProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        CollapseByDefault: new MockTwoOptionsProperty(),
        InputEvent: new MockStringProperty(),
        SelectedKey: new MockStringProperty(),
        Theme: new MockStringProperty(),
        items: new MockDataSet([]),
    };
}
