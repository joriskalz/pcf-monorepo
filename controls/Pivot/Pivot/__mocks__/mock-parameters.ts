/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockEnumProperty, MockStringProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        RenderSize: new MockEnumProperty(),
        RenderType: new MockEnumProperty(),
        SelectedKey: new MockStringProperty(),
        Theme: new MockStringProperty(),
        items: new MockDataSet([]),
    };
}
