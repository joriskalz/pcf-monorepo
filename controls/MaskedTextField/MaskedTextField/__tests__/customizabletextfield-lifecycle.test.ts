import { MaskedTextField } from '..';
import { IInputs } from '../generated/ManifestTypes';
import { MockContext, MockState } from '../__mocks__/mock-context';
import { getMockParameters } from '../__mocks__/mock-parameters';
import * as renderer from 'react-test-renderer';
import { mount } from 'enzyme';

describe('MaskedTextField', () => {
    beforeEach(() => jest.clearAllMocks());
    afterEach(() => {
        for (let i = 0; i < document.body.children.length; i++) {
            if (document.body.children[i].tagName === 'DIV') {
                document.body.removeChild(document.body.children[i]);
                i--;
            }
        }
    });

    it('renders', () => {
        const { component, context, notifyOutputChanged } = createComponent();
        component.init(context, notifyOutputChanged);
        const element = component.updateView(context);
        expect(element).toMatchSnapshot();
    });

    it('on text enter', () => {
        const { component, context, notifyOutputChanged } = createComponent();
        const valueText = 'Hello';
        component.init(context, notifyOutputChanged);
        const textboxComponent = component.updateView(context);
        const textboxField = mount(textboxComponent);
        const textboxFieldElement = textboxField.find('.ms-TextField-field').first();
        expect(textboxFieldElement.length).toEqual(1);
        textboxFieldElement.simulate('change', { target: { value: valueText } });
        const outputs = component.getOutputs();
        expect(outputs.Value).toEqual(valueText);
    });

    it('theme', async () => {
        const { component, context, notifyOutputChanged } = createComponent();
        context.parameters.Theme.raw = JSON.stringify({
            palette: {
                themePrimary: '#0078d4',
            },
        });
        context.parameters.Value.raw = 'Secret';
        component.init(context, notifyOutputChanged);
        const customizableInput = renderer.create(component.updateView(context));
        expect(customizableInput.toJSON()).toMatchSnapshot();
    });
});

function createComponent() {
    const component = new MaskedTextField();
    const context = new MockContext<IInputs>(getMockParameters());
    context.parameters.Prefix.raw = 'pre_';
    context.parameters.Suffix.raw = '.com';
    context.parameters.ErrorMessage.raw = 'an error occured';
    context.parameters.Mask.raw = '************************';
    context.parameters.MaskFormat.raw = '[a-zA-Z0-9]';
    const notifyOutputChanged = jest.fn();
    const state = new MockState();
    const container = document.createElement('div');
    document.body.appendChild(container);
    return { component, context, notifyOutputChanged, container, state };
}
