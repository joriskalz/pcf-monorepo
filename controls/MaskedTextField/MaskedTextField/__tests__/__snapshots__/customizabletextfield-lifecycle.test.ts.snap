// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MaskedTextField renders 1`] = `
<Memo(CanvasMaskedTextField)
  ariaLabel=""
  disabled={false}
  errorMessage="an error occured"
  mask="************************"
  maskChar=""
  maskPattern="[a-zA-Z0-9]"
  onChange={[Function]}
  prefixValue="pre_"
  setFocus=""
  suffixValue=".com"
  themeJSON=""
  value=""
  width={-1}
/>
`;

exports[`MaskedTextField theme 1`] = `
<div
  className=

      {
        -moz-osx-font-smoothing: grayscale;
        -webkit-font-smoothing: antialiased;
        background: #ffffff;
        color: #323130;
        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        font-size: 14px;
        font-weight: 400;
      }
>
  <div
    className=
        ms-TextField
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          box-shadow: none;
          box-sizing: border-box;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          margin-bottom: 0px;
          margin-left: 0px;
          margin-right: 0px;
          margin-top: 0px;
          padding-bottom: 0px;
          padding-left: 0px;
          padding-right: 0px;
          padding-top: 0px;
          position: relative;
          width: -1px;
        }
  >
    <div
      className="ms-TextField-wrapper"
    >
      <div
        className=
            ms-TextField-fieldGroup
            {
              align-items: stretch;
              background: #ffffff;
              border-color: #a4262c;
              border-radius: 2px;
              border: 1px solid #605e5c;
              box-shadow: none;
              box-sizing: border-box;
              cursor: text;
              display: flex;
              flex-direction: row;
              height: 32px;
              margin-bottom: 0px;
              margin-left: 0px;
              margin-right: 0px;
              margin-top: 0px;
              padding-bottom: 0px;
              padding-left: 0px;
              padding-right: 0px;
              padding-top: 0px;
              position: relative;
            }
            &:hover {
              border-color: #a4262c;
            }
            @media screen and (-ms-high-contrast: active), screen and (forced-colors: active){&:hover {
              -ms-high-contrast-adjust: none;
              border-color: Highlight;
              forced-color-adjust: none;
            }
      >
        <div
          className=
              ms-TextField-prefix
              {
                align-items: center;
                background: #f3f2f1;
                color: #605e5c;
                display: flex;
                flex-shrink: 0;
                line-height: 1px;
                padding-bottom: 0;
                padding-left: 10px;
                padding-right: 10px;
                padding-top: 0;
                white-space: nowrap;
              }
              @media screen and (-ms-high-contrast: active), screen and (forced-colors: active){& {
                background: Window;
                color: WindowText;
              }
        >
          <span
            style={
              Object {
                "paddingBottom": "1px",
              }
            }
          >
            pre_
          </span>
        </div>
        <input
          aria-describedby="TextFieldDescription6"
          aria-invalid={true}
          aria-label=""
          className=
              ms-TextField-field
              {
                -moz-osx-font-smoothing: grayscale;
                -webkit-font-smoothing: antialiased;
                background-color: transparent;
                background: none;
                border-radius: 0px;
                border: none;
                box-shadow: none;
                box-sizing: border-box;
                color: #323130;
                font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
                font-size: 14px;
                font-weight: 400;
                margin-bottom: 0px;
                margin-left: 0px;
                margin-right: 0px;
                margin-top: 0px;
                min-width: 0px;
                outline: 0px;
                padding-bottom: 0;
                padding-left: 8px;
                padding-right: 8px;
                padding-top: 0;
                text-overflow: ellipsis;
                width: 100%;
              }
              &:active {
                outline: 0px;
              }
              &:focus {
                outline: 0px;
              }
              &:hover {
                outline: 0px;
              }
              &::-ms-clear {
                display: none;
              }
              @media screen and (-ms-high-contrast: active), screen and (forced-colors: active){& {
                background: Window;
                color: WindowText;
              }
              &::placeholder {
                -moz-osx-font-smoothing: grayscale;
                -webkit-font-smoothing: antialiased;
                color: #605e5c;
                font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
                font-size: 14px;
                font-weight: 400;
                opacity: 1;
              }
              @media screen and (-ms-high-contrast: active), screen and (forced-colors: active){&::placeholder {
                color: GrayText;
              }
              &:-ms-input-placeholder {
                -moz-osx-font-smoothing: grayscale;
                -webkit-font-smoothing: antialiased;
                color: #605e5c;
                font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
                font-size: 14px;
                font-weight: 400;
                opacity: 1;
              }
              @media screen and (-ms-high-contrast: active), screen and (forced-colors: active){&:-ms-input-placeholder {
                color: GrayText;
              }
              &::-ms-input-placeholder {
                -moz-osx-font-smoothing: grayscale;
                -webkit-font-smoothing: antialiased;
                color: #605e5c;
                font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
                font-size: 14px;
                font-weight: 400;
                opacity: 1;
              }
              @media screen and (-ms-high-contrast: active), screen and (forced-colors: active){&::-ms-input-placeholder {
                color: GrayText;
              }
          disabled={false}
          id="TextField5"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onInput={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          onPaste={[Function]}
          type="text"
          value=""
          width={-1}
        />
        <div
          className=
              ms-TextField-suffix
              {
                align-items: center;
                background: #f3f2f1;
                color: #605e5c;
                display: flex;
                flex-shrink: 0;
                line-height: 1px;
                padding-bottom: 0;
                padding-left: 10px;
                padding-right: 10px;
                padding-top: 0;
                white-space: nowrap;
              }
              @media screen and (-ms-high-contrast: active), screen and (forced-colors: active){& {
                background: Window;
                color: WindowText;
              }
        >
          <span
            style={
              Object {
                "paddingBottom": "1px",
              }
            }
          >
            .com
          </span>
        </div>
      </div>
    </div>
    <span
      id="TextFieldDescription6"
    >
      <div
        role="alert"
      />
    </span>
  </div>
</div>
`;
